R/labs_to_levels.R

```diff
@@ -0,0 +1,116 @@
+#' @title Convert value labels into factor levels
+#' @name labs_to_levels
+#'
+#' @details
+#' Convert variables or data into factors and uses value labels as factor levels.
+#' This function is similar to `to_factor()`, however, it is specially designed
+#' for factors with value labels attributes. While `to_factor()` returns factors
+#' as is, `labs_to_levels()` always uses value labels as factor levels, even
+#' for factors.
+#'
+#' @param x A data frame or vector.
+#' @param remove_attr Logical, if `"labels"` attribute should be removed after
+#' value labels are set as factor levels.
+#' @param ... Arguments passed to or from other methods, in this in particular
+#' to `to_factor()`.
+#' @inheritParams find_columns
+#' @inheritParams categorize
+#'
+#' @return A factor, or a data frame of factors.
+#'
+#' @examples
+#' str(labs_to_levels(iris))
+#'
```

Comment:
This example is weird to me and actually I think `labs_to_levels()` does too much. It doesn't *only* converts labels to levels, it also converts to factor (which makes sense since we're talking about factors) but I think a function should do one thing only.

If someone wants to convert to factor, they should use `to_factor()`, potentially with the option to convert labs to levels.

This is what I have in mind:

* I have a labelled factor but labels are not levels -> `labs_to_levels()`
* I have an unlabelled factor -> `labs_to_levels()` errors 
* for other things, `to_factor(labs_to_levels = TRUE)` or `to_factor(labs_to_levels = FALSE)` + `labs_to_levels()`

Reply:
Should we then better add this as an option to `to_factor()`, instead of having two functions? Or would it be weird that `to_factor()` actually changes factors when argument `labs_to_levels = TRUE`?

Reply:
but then in the first case (labelled factor but labels are not levels), you would need to pass a factor into `to_factor()` so that would be weird. I think this `labs_to_levels()` (or whatever the name is) can exist but its goal should be limited to only put labels as levels